Panic during Kaniko build

I was running `skaffold build` (using Kaniko), and looking at event logs, it seems like a node issue caused the build to fail with a panic. I don't have enough logging to be sure how to reproduce, although hopefully the [code reference](https://github.com/GoogleContainerTools/skaffold/blob/1bbf9a3f13094e58c90399a4bd7e07250e0d2da5/pkg/skaffold/kubernetes/wait.go#L93) might be enough...

### Expected behavior
Error message when watching the Kaniko pod failed

### Actual behavior
```
panic: interface conversion: runtime.Object is nil, not *v1.Pod

goroutine 32 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes.WaitForPodComplete.func1(0xc0000d0000, 0xc0000d0000, 0x2, 0x1)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/wait.go:93 +0x389
github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes.watchUntil(0x17cec60, 0xc0000aa540, 0x17baba0, 0xc000558e70, 0xc000067bd8, 0xc0005488b0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/wait.go:45 +0x173
github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes.WaitForPodComplete(0x17cec60, 0xc0000aa540, 0x17e6440, 0xc00038d1e0, 0xc000447360, 0xc, 0x2ba7def3000, 0x0, 0x0)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/wait.go:92 +0x2f6
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/kaniko.(*Builder).run(0xc000553040, 0x17cebe0, 0xc00033ac00, 0x17b1900, 0xc00055c4d8, 0xc0004ed140, 0xc000596000, 0x4a, 0x0, 0x0, ...)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/kaniko/run.go:97 +0x67d
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/kaniko.(*Builder).buildArtifactWithKaniko(0xc000553040, 0x17cebe0, 0xc00033ac00, 0x17b1900, 0xc00055c4d8, 0xc0004ed140, 0xc000596000, 0x4a, 0x143fa20, 0x1455e80, ...)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/kaniko/kaniko.go:49 +0x9b
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/kaniko.(*Builder).buildArtifactWithKaniko-fm(0x17cebe0, 0xc00033ac00, 0x17b1900, 0xc00055c4d8, 0xc0004ed140, 0xc000596000, 0x4a, 0x1, 0x47, 0x0, ...)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/kaniko/kaniko.go:45 +0x83
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build.InParallel.func1(0x17b1ea0, 0xc0000a8008, 0xc00055c4d8, 0xc00054a660, 0x2, 0x2, 0x0, 0xc000556720, 0xc0000b4780, 0x2, ...)
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/parallel.go:77 +0x28e
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/build.InParallel
	/go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/parallel.go:59 +0x2d0
```

### Information

- Skaffold version: v0.28.0
- Operating system: Ubuntu 18.04

--- a/pkg/skaffold/kubernetes/wait.go
+++ b/pkg/skaffold/kubernetes/wait.go
@@ -90,6 +90,9 @@ func WaitForPodComplete(ctx context.Context, pods corev1.PodInterface, podName s
 	defer cancelTimeout()
 
 	return watchUntil(ctx, w, func(event *watch.Event) (bool, error) {
+		if event.Object == nil {
+			return false, nil
+		}
 		pod := event.Object.(*v1.Pod)
 		if pod.Name != podName {
 			return false, nil

